onnxruntime/core/graph/graph.cc

```diff
@@ -2511,15 +2552,8 @@ Status Graph::VerifyNodeAndOpMatch(const ResolveOptions& options) {
       if (node.since_version_ == -1) {
         node.since_version_ = node.op_->since_version();
       }
-    } else {
-      // This is only applicable for model local functions.
-      // In case of nested model local functions, graph resolve is called during resolve for parent
-      // function body graph otherwise type inference for nest function cannot happen.
-      if (options.traverse_function_body && node.GetFunctionBody() != nullptr) {
```

Comment:
This is a design question to think about (and also document in ONNX ... this could also impact the ONNX checker. It would be useful to ensure we do the same in both places). If a model-local function has an error (eg., type-error), do we want to ignore it in case we use a kernel?

Reply:
Maybe ensure that the ONNX checker has an option for this.

Reply:
i thought we run the checker before assign op schema to the nodes?

Reply:
I am not sure what exactly gets checked. Because type-and-shape-inference check is extra, on top of basic checker. In the original implementation, ORT did the top-level type-and-shape-inference itself (calling only node-level inference methods). Would be good to ensure that it gets checked one place or another.